setup_wordpress.sh

```diff
@@ -1,3 +1,245 @@
+sudo chown -R $USER:$USER /var/www/wordpress
+read -p "Enter the public IP address, please (192.168.X.X): " ip_address
+
+echo "
+server {
+    listen 80;
+    server_name $ip_address;
+    root /var/www/wordpress;
+
+    index index.html index.htm index.php;
+
+    location / {
+        try_files \$uri \$uri/ =404;
+    }
+
+    location ~ \.php$ {
+        include snippets/fastcgi-php.conf;
+        fastcgi_pass unix:/var/run/php/php7.4-fpm.sock;
+    }
+
+    location ~ /\.ht {
+        deny all;
+    }
+
+}" | sudo tee /etc/nginx/sites-available/wordpress
+
+echo "------------------ Creating Soft Link ------------------"
+sudo ln -s /etc/nginx/sites-available/wordpress /etc/nginx/sites-enabled/
+sudo ln -s /etc/nginx/sites-available/wordpress /etc/nginx/sites-enabled/
+
+echo "------------------ Testing Nginx ------------------"
+sudo nginx -t
+
+echo "------------------ Restarting Nginx ------------------"
+sudo systemctl reload nginx
+
+echo "------------------ Creating Nginx Index ------------------"
+echo "
+<html>
+  <head>
+    <title>Rojin's Website</title>
+  </head>
+  <body>
+    <h1>Hello Rojin!</h1>
+  </body>
+</html>" | sudo tee /var/www/wordpress/index.html
+
+echo "You Can Test The Index Page At ${ip_address}"
+
+echo "------------------ Creating SSL Certificate ------------------"
+sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/ssl/private/nginx-selfsigned.key -out /etc/ssl/certs/nginx-selfsigned.crt
+sudo openssl dhparam -out /etc/nginx/dhparam.pem 2048
+
+echo "------------------ Creating Configuration Snippet ------------------"
+sudo echo "
+ssl_certificate /etc/ssl/certs/nginx-selfsigned.crt;
+ssl_certificate_key /etc/ssl/private/nginx-selfsigned.key;" | sudo tee /etc/nginx/snippets/self-signed.conf
+
+echo "------------------ Creating ssl-params.conf ------------------"
+sudo echo '
+ssl_protocols TLSv1.2;
+ssl_prefer_server_ciphers on;
+ssl_dhparam /etc/nginx/dhparam.pem;
+ssl_ciphers ECDHE-RSA-AES256-GCM-SHA512:DHE-RSA-AES256-GCM-SHA512:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384;
+ssl_ecdh_curve secp384r1; # Requires nginx >= 1.1.0
+ssl_session_timeout  10m;
+ssl_session_cache shared:SSL:10m;
+ssl_session_tickets off; # Requires nginx >= 1.5.9
+# ssl_stapling on; # Requires nginx >= 1.3.7
+# ssl_stapling_verify on; # Requires nginx => 1.3.7
+resolver 8.8.8.8 8.8.4.4 valid=300s;
+resolver_timeout 5s;
+# Disable strict transport security for now. You can uncomment the following
+# line if you understand the implications.
+# add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload";
+add_header X-Frame-Options DENY;
+add_header X-Content-Type-Options nosniff;
+add_header X-XSS-Protection "1; mode=block";' | sudo tee /etc/nginx/snippets/ssl-params.conf
+
+echo "------------------ Adjusting Nginx To SSL ------------------"
+sudo cp /etc/nginx/sites-available/wordpress /etc/nginx/sites-available/wordpress.bak
+
+echo "
+server {
```

Comment:
why do you need to rerender nginx configuration file for wordpress multiple times? Can it be configured as it should like in the end?

Reply:
I thought maybe I could test if the Nginx was ok and running after a simple configuration, but I did not add it. So yes, I need to remove the unnecessary parts.